packages/builder/src/components/backend/DataTable/modals/grid/GridCreateColumnModal.svelte

```diff
@@ -2,7 +2,13 @@
   import { getContext } from "svelte"
   import CreateEditColumn from "components/backend/DataTable/modals/CreateEditColumn.svelte"
 
-  const { datasource } = getContext("grid")
+  const { datasource, rows } = getContext("grid")
 </script>
 
-<CreateEditColumn on:updatecolumns={datasource.actions.refreshDefinition} />
+<CreateEditColumn
+  on:updatecolumns={async () => {
+    await datasource.actions.refreshDefinition()
+    // Need to find a better way to handle formula columns than just always refreshing
+    rows.actions.refreshData()
```

Comment:
I would prefer to leave this out and try to find some other way. Not only will this increase API load substantially, it will also constantly reset the users current scroll positions and paginated data. We could maybe add a unique event for when formula columns are created/changed and use that to trigger data refreshes?